### ISSUE TITLE
Training Epoch

### ISSUE THREAD

**Issue by keelezibel** [Rongjiehuang/ProDiff#11]
Hi thanks for the repo, wanted to check in with the training metrics. Currently, i am training the ProDiff Teacher model and getting the following validation results. Saw that there are a couple of metrics. 

```
Valid results: {'total_loss': 1.4424, 'ssim': 0.199, 'l1': 0.1656, 'pdur': 0.0264, 'wdur': 0.058, 'sdur': 0.0241, 'uv': 0.685, 'f0': 0.2621, 'e': 0.0222
```

- Which metrics should i note if there is an improvement over time? 
- What is a reasonable loss for finetuning? I am currently running off the checkpoints of the pre-trained model provided by you guys.
- Also, noted that after training the ProDiff teacher model, i need to run the training using ProDiff yaml. Is there any chance we need to train the FastDiff model too? 

Thanks a lot!

**Comment by Rongjiehuang** [comment]
Hi, you may find `l1` in training schedules useful to judge the learning or fine-tuning, and please do not use the metrics in validation schedules. Secondly, it would be fine to tune the fastdiff model too, which will promote the audio quality.

**Comment by keelezibel** [comment]
Hi @Rongjiehuang, I am back with some results better than the last run.

[Reference Audio](https://on.soundcloud.com/gyjhC) 
[Synthesized Audio](https://on.soundcloud.com/x5Mhg)

I have trained this model up to 26k steps by finetuning on your released ProDiff-Teacher model. The current L1 training loss is 0.0107. I noticed there are still artefacts in the synthesized voice compared to the reference audio. Currently, I am training on a relatively smaller dataset of around 20mins compared to ~20H for LJSpeech.

I am thinking there could be several options here:
1. Increase the number of training epochs 
2. Increase dataset size
3. Finetune the Prodiff as well
4. Look into how to finetune the fastdiff model as you had mentioned

Thanks alot for releasing this repo.

**Comment by AIFahim** [comment]
![image](https://github.com/Rongjiehuang/ProDiff/assets/33654834/d3a55429-2d19-4be1-9b9c-de834a2c7cf4)
@keelezibel I got the total_loss = 0! Did you also get the training loss 0?

**Comment by keelezibel** [comment]
@AIFahim no i didn't get zero for training loss. You can refer to my first screenshot. I eventually settled for PaddleSpeech TTS as I couldnt figure out how to improve the quality of the voice further. Also, there were too many dependencies on other existing projects.

**Comment by 1208320416** [comment]
@AIFahim Hi，I have encountered the same problem as you. I think there may be a problem with my data preprocessing, because I found that the dateloader is empty. Have you solved the problem yet?
